src/main/java/org/jabref/logic/importer/util/GrobidService.java

```diff
@@ -45,18 +52,39 @@ public GrobidService(String grobidServerURL) {
      * @throws IOException if an I/O excecption during the call ocurred or no BibTeX entry could be determiend
      */
     public String processCitation(String rawCitation, ConsolidateCitations consolidateCitations) throws IOException {
-        rawCitation = URLEncoder.encode(rawCitation, StandardCharsets.UTF_8);
-        URLDownload urlDownload = new URLDownload(grobidServerURL
-                + "/api/processCitation");
-        urlDownload.setConnectTimeout(Duration.ofSeconds(5));
-        urlDownload.addHeader("Accept", MediaTypes.APPLICATION_BIBTEX);
-        urlDownload.setPostData("citations=" + rawCitation + "&consolidateCitations=" + consolidateCitations);
-        String httpResponse = urlDownload.asString();
+        Connection.Response response = Jsoup.connect(grobidServerURL + "/api/processCitation")
+                .header("Accept", MediaTypes.APPLICATION_BIBTEX)
+                .data("citations", rawCitation)
+                .data("consolidateCitations", String.valueOf(consolidateCitations.getCode()))
+                .method(Connection.Method.POST)
+                .ignoreContentType(true)
+                .execute();
+        String httpResponse = response.body();
```

Comment:
In URLDownload we have some logic about bypassing SSL certifications. Something to think about here with Jsoup?

Reply:
I don't think it's necessary as the Grobid Server is controlled by JabRef and there is no SSL certificate in place.